### ISSUE TITLE
entrenar: D2D buffer size mismatch in CudaTransformerBlock backward_attention

### ISSUE THREAD

**Issue by noahgift** [paiml/albor#39]
## Bug

`CudaTransformerBlock::backward_attention()` uses `gate_out` (sized `seq_len * intermediate_size`) as a temp buffer for `residual_add_forward` during K/V gradient accumulation, then copies to `grad_hidden` (sized `seq_len * hidden_size`) via `copy_from_buffer_async`. Since `intermediate_size != hidden_size`, the D2D copy fails with:

```
PAR-023: Async D2D length mismatch: src 262144 vs dst 65536
```

## Root Cause

`gate_out` is `seq_len * intermediate_size` (e.g., 128 * 2048 = 262144). `grad_hidden` is `seq_len * hidden_size` (e.g., 128 * 512 = 65536). The `copy_from_buffer_async` requires exact length match.

## Fix

Use `o_proj_out` (sized `seq_len * hidden_size`, same as `grad_hidden`) as the temp buffer instead. At this point in the backward pass, `o_proj_out` is no longer needed.

Fixed in `entrenar` — both K and V accumulation paths.

## Impact

Blocks all GPU-resident pretraining via `CudaTransformerTrainer` (ALB-040). The forward pass worked but backward silently failed, causing loss=0.0 on every step.

Refs #38

**Comment by noahgift** [comment]
Fixed in entrenar@a48e3d2 (ALB-041). D2D buffer size uses max_seq_len for pre-allocated buffers with seq truncation handling.
